### PR TITLE
Enable Style/RedundantSelf Rubocop cop

### LINKED ISSUE #1373: Enable Style/RedundantSelf Rubocop cop
There are many redundant selfs in the code. Let's enable the Rubocop cop for that!

https://rubocop.readthedocs.io/en/latest/cops_style/#styleredundantself

### PR DESCRIPTION
**Pull request by BelieveC**
This cop checks for redundant uses of self. It supports autocorrection, so all the offenses where automatically solved.

Closes #1373

### CODE REVIEW

**Comment by BelieveC** [comment]
Alright! :+1:

**Comment by Ana06** [comment]
https://github.com/openSUSE/osem/pull/1400 got merged so this can get merged now. @BelieveC  please rebase! :wink:

**Comment by BelieveC** [comment]
On it :+1:

**Comment by BelieveC** [comment]
Done :+1:

File: .rubocop.yml [hunk 1]
```diff
@@ -88,6 +88,10 @@ Style/TrailingBlankLines:
 Style/TrailingWhitespace:
   Enabled: true
 
+# Checks for redundant uses of self.
```
Comment (Ana06) [inline .rubocop.yml]: This is alphabetically ordered, so it should be in the  79, before `Style/StringLiterals`

Comment (BelieveC) [inline .rubocop.yml]: Gotcha! I will make the changes :+1:

**Comment by BelieveC** [comment]
@Ana06 Done :+1:

**Pull request merged**